packages/frontend/src/hooks/user/useUserDeleteMutation.ts

```diff
@@ -0,0 +1,15 @@
+import { ApiError } from '@lightdash/common';
+import { useMutation } from 'react-query';
+import { lightdashApi } from '../../api';
+
+const deleteUserQuery = async () =>
+    lightdashApi<undefined>({
+        url: `/user/me`,
+        method: 'DELETE',
+        body: undefined,
+    });
+
+export const useDeleteUserMutation = () =>
```

Comment:
have another issue that will focus on renaming another existing mutation with the same name: 
https://github.com/lightdash/lightdash/issues/5139